R/further_analytical_functions.R

```diff
@@ -0,0 +1,308 @@
+  q1_0_range <- paste0(round(q1_0 - 0.016,3), "--", round(q1_0 + 0.016,3))
+
+  # corrected mortality for age group 5q0 after Bocquet-Appel and Masset
+  q5_0 <- 1.154 * sqrt((log10(200 * indx$juvenile_i$d5_14_d20plus))) - 1.014
+  q5_0_range <- paste0(round(q5_0 - 0.041,3), "--", round(q5_0 + 0.041,3))
+
+  Dx_sum_corrected <- (life_table$Dx %>% sum - life_table$Dx[1]) / (1 - q5_0)
+  Dx5_0_corrected <- q5_0 * Dx_sum_corrected
+
+  if (5 == life_table$a[1]) {
+    life_table$Dx[[1]] <- Dx5_0_corrected
+  } else if ((4 == life_table$a[2]) & (1 == life_table$a[1])) {
+    Dx1_0_corrected <- q1_0 * Dx_sum_corrected
+    life_table$Dx[[1]] <- Dx1_0_corrected
+    life_table$Dx[[2]] <- Dx5_0_corrected - Dx1_0_corrected
+  } else {
+    stop("Life table correction works only with one 5-year-age class or 1- and 4-year classes
+         for the first 5 years. Please take a look at ?life.table to determine how your
+         input data should look like for accomplishing this.")
+  }
+  life_table_prep_corrected <- data.frame(cbind(a = life_table$a, Dx = life_table$Dx))
+  life_table_corr <- life.table(life_table_prep_corrected, agecor = agecor, agecorfac = agecorfac, option_spline = option_spline)
+  row_e0 <- c(round(e0, 1), e0_range)
+  row_q1_0 <- c(round(q1_0, 3), q1_0_range)
+  row_q5_0 <- c(round(q5_0, 3), q5_0_range)
+  e0_q5_0 <- data.frame(rbind(row_e0, row_q1_0, row_q5_0))
+  colnames(e0_q5_0) <- c("value", "range")
+  rownames(e0_q5_0) <- c("e0", "1q0", "5q0")
+  return(list(indices = e0_q5_0, life_table_corr = life_table_corr))
+  }
+
+
+#' Calculate masculinity index and maternal mortality
+#' from a mortAAR life table list
+#'
+#' The proportional relation between adult males and females (=
+#' Masculinity index) is interesting for a number of reasons: (1)
+#' it can point to basic problems in the datasets in that, say,
+#' one sex is grossly over- or underrepresented. (2) it may hint
+#' towards cultural reaons like sex-specific mobility.\cr
+#' Maternal mortality is a basic indicator for the health system of a
+#' given population. Recently, McFadden and Oxenham (2019) have
+#' provided a formula to calculate it from archaeological data.
+#'
+#' The Masculinity index (MI) is defined for juvenile and older
+#' individuals. Note that with a higher mortality rate of adult females,
+#' an MI < 100 does not necesssarily speak for an unbalanced MI in life.\cr
+#' Maternal mortality is calculated according to the formula
+#' provided by McFadden and Oxenham (2019). McFadden and Oxenham show
+#' that with modern data a very high correlation is achieved by only
+#' comparing the absolute numbers of the age group 20 to 24. This has the
+#' additional advantage that for this age group anthropological aging
+#' methods are reasonable exact.
+#'
+#' @param life_table an object of class mortaar_life_table_list.
+#' @param females an object of class mortaar_life_table for females
```

Comment:
This is just the group name as a character. Not the object. 

Maybe it's better to change the input: No life.table.list which contains the two life.tables + the relevant names, but just the two relevant life.tables.

Reply:
Done.

Reply:
Nice! But do we still need the first argument, the `mortaar_life_table_list`, in this case?

Reply:
No, it is deleted in the last commit.